lib/gems/pending/util/miq-password.rb

```diff
@@ -185,16 +186,79 @@ def self.ez_load(filename, recent = true)
     if !File.exist?(filename)
       nil
     elsif recent
-      EzCrypto::Key.load(filename)
+      params = YAML.load_file(filename)
+      Key.new(*params.values_at(:algorithm, :key, :iv))
     else
       params = YAML.load_file(filename)
-      CryptString.new(nil, params[:algorithm], params[:key], params[:iv])
+      algorithm, key, iv = params.values_at(:algorithm, :key, :iv)
+      Key.new(algorithm, key && Base64.encode64(key), iv && Base64.encode64(iv))
     end
   end
 
   def self.extract_erb_encrypted_value(value)
     return $1 if value =~ /\A<%= (?:MiqPassword|DB_PASSWORD)\.decrypt\(['"]([^'"]+)['"]\) %>\Z/
   end
+
+  class Key
+    GENERATED_KEY_SIZE = 32
+
+    def self.generate_key(password = nil, salt = nil)
+      password ||= OpenSSL::Random.random_bytes(GENERATED_KEY_SIZE)
+      Base64.encode64(Digest::SHA256.digest("#{password}#{salt}")[0, GENERATED_KEY_SIZE]).chomp
+    end
+
+    def initialize(algorithm = nil, key = nil, iv = nil)
+      @algorithm = algorithm || "aes-256-cbc"
+      @key       = key || generate_key
+      @raw_key   = Base64.decode64(@key)
+      @iv        = iv
+      @raw_iv    = iv && Base64.decode64(iv)
+    end
+
+    def encrypt(str)
+      apply(:encrypt, str)
+    end
+
+    def encrypt64(str)
+      Base64.encode64(encrypt(str)).chomp
```

Comment:
would using `strict_encode64` work for us here?

Think there may have been a reason I avoided in the past, but so much has changed since then...